File: repertoire.server/api/requests/song_part_requests.go
package requests

import "github.com/google/uuid"

type CreateSongPartRequest struct {
	SongID       uuid.UUID `validate:"required"`
	Name         string    `validate:"required,max=30"`
	SectionIDs   []uuid.UUID
	BandMemberID *uuid.UUID
	InstrumentID *uuid.UUID
}

type UpdateSongPartRequest struct {
	ID           uuid.UUID `validate:"required"`
	Name         string    `validate:"required,max=30"`
	Confidence   uint      `validate:"max=100"`
	Rehearsals   uint
	BandMemberID *uuid.UUID
	InstrumentID *uuid.UUID
}

type UpdateAllSongPartsRequest struct {
	SongID       uuid.UUID `validate:"required"`
	InstrumentID *uuid.UUID
	BandMemberID *uuid.UUID
}

type MoveSongPartInSongRequest struct {
	ID     uuid.UUID `validate:"required"`
	OverID uuid.UUID `validate:"required"`
	SongID uuid.UUID `validate:"required"`
}

type MoveSongPartInSectionRequest struct {
	ID        uuid.UUID `validate:"required"`
	OverID    uuid.UUID `validate:"required"`
	SectionID uuid.UUID `validate:"required"`
}

type BulkUpdateSongPartsRequest struct {
	Requests []BulkUpdateSongPartRequest `validate:"min=1,dive"`
	SongID   uuid.UUID                   `validate:"required"`
}

type BulkDeleteSongPartsRequest struct {
	IDs    []uuid.UUID `validate:"min=1"`
	SongID uuid.UUID   `validate:"required"`
}

type BulkUpdateSongPartRequest struct {
	ID         uuid.UUID `validate:"required"`
	Confidence uint      `validate:"max=100"`
	Rehearsals uint
}

File: repertoire.ui/src/features/songs/features/song/features/parts/types/requests/SongPartRequests.ts
export interface GetSongPartsRequest {
  songId: string
}

export interface CreateSongPartRequest {
  songId: string
  name: string
  instrumentId?: string
  sectionId?: string
  bandMemberId?: string
}

export interface BulkUpdateSongPartsRequest {
  requests: { id: string; rehearsals: number, confidence: number }[]
  songId: string
}

export interface UpdateSongPartRequest {
  id: string
  name: string
  rehearsals: number
  confidence: number
  instrumentId?: string
  sectionId?: string
  bandMemberId?: string
}

export interface UpdateAllSongPartsRequest {
  songId: string
  bandMemberId?: string
  instrumentId?: string
}

export interface MoveSongPartInSongRequest {
  id: string
  overId: string
  songId: string
}

export interface BulkDeleteSongPartsRequest {
  ids: string[]
  songId: string
}

export interface DeleteSongPartRequest {
  id: string
  songId: string
}
